### PR TITLE
Kafka robustness 2

### PR DESCRIPTION
**Pull request by JKL98ISR**

### CODE REVIEW

File: backend/deepchecks_monitoring/logic/data_ingestion.py [hunk 1]
```diff
@@ -359,20 +359,29 @@ async def _send_with_retry(
...
 
             failed_messages = [
-                (key, message) for ((key, message), result) in zip(messages, results)
+                ((key, message), result) for ((key, message), result) in zip(messages, results)
                 if isinstance(result, Exception)
             ]
 
+            exception_types = {type(result).__name__ for _, result in failed_messages}
+            if len(exception_types) == 1:
+                self.logger.warning(f"All messages failed with the same error: {list(exception_types)[0]}")
```
Comment (shayts7) [inline backend/deepchecks_monitoring/logic/data_ingestion.py:375]: why not error?

File: backend/deepchecks_monitoring/logic/data_ingestion.py [hunk 2]
```diff
@@ -359,20 +359,29 @@ async def _send_with_retry(
...
+            else:
+                for ((key, _), result) in failed_messages:
+                    self.logger.warning(
```
Comment (shayts7) [inline backend/deepchecks_monitoring/logic/data_ingestion.py:378]: Also here, why not error?

**Review by shayts7 — APPROVED** [review]
See comments inside

File: backend/deepchecks_monitoring/logic/data_ingestion.py (hunk 1 above)
Comment (JKL98ISR) [reply backend/deepchecks_monitoring/logic/data_ingestion.py:375]: there's a retry and an error only if failed after retries

File: backend/deepchecks_monitoring/logic/data_ingestion.py (hunk 2 above)
Comment (JKL98ISR) [reply backend/deepchecks_monitoring/logic/data_ingestion.py:378]: there's a retry and an error only if failed after retries

**Pull request merged**